src/reyna/Connection/ConnectionInfo.cs

```diff
@@ -0,0 +1,70 @@
+﻿namespace Reyna
+{
+    using System;
+    using System.IO;
+    using OpenNETCF.Net.NetworkInformation;
+
+    public class ConnectionInfo : IConnectionInfo
+    {
+        public bool Connected
+        {
+            get
+            {
+                try
+                {
+                    foreach (var ni in NetworkInterface.GetAllNetworkInterfaces())
+                    {
+                        if (string.Compare(ni.CurrentIpAddress.ToString().Trim(), "0.0.0.0", true) != 0)
+                        {
+                            return true;
+                        }
+                    }
+                }
+                catch (Exception)
+                { 
+                }
+                
+                return false;
+            }
+        }
+
+        public bool Mobile
+        {
+            get
+            {
+                try
+                {
+                    var interfaces = NetworkInterface.GetAllNetworkInterfaces();
+                    bool mobileNetworkConnected = false;
+                    bool otherNetworksConnected = false;
+                    foreach (var ni in interfaces)
+                    {
+                        if (GPRSNetwork(ni))
+                        {
+                            if (string.Compare(ni.CurrentIpAddress.ToString().Trim(), "0.0.0.0", true) != 0)
+                            {
+                                mobileNetworkConnected = true;
+                            }
+                        }
+                        else if (string.Compare(ni.CurrentIpAddress.ToString().Trim(), "0.0.0.0", true) != 0)
+                        {
+                            otherNetworksConnected = true;
```

Comment:
I need to check if there is another connection available, as device can be connected to LAN, WLAN, USB data and mobile then in this case we return false. it is not mobile only. then we leave it to the system to determine which connection will use to transfer data.

@Youhana-Hana